### PR TITLE
bpo-38870: Implement round tripping support for typed AST

### PR DESCRIPTION
**Pull request by isidentical**
https://bugs.python.org/issue38870

### CODE REVIEW

**Comment by pablogsal** [comment]
@isidentical Can you rebase this against the latest master? Ping me when is ready and I can do another review

**Comment by isidentical** [comment]
Rebased @pablogsal

File: Lib/ast.py [hunk 1]
```diff
@@ -666,11 +667,16 @@ def buffer(self):
         return value
 
     @contextmanager
-    def block(self):
+    def block(self, node=None, *, append_type_comment = False):
```
Comment (pablogsal) [inline Lib/ast.py]: This function is doing too much.

 Its responsibility is managing the source for blocks, it should not need to understand or deal with type comments

**Review by pablogsal — CHANGES_REQUESTED** [review]

File: Lib/ast.py (hunk 1 above)
Comment (isidentical) [reply Lib/ast.py]: There is no way we can intervene between colon and the new line after that.

Comment (pablogsal) [reply Lib/ast.py]: Yes, the function can receive an optional, already calculated, type comment and then it blindly appends it. This way the logic becomes uncoupled and the "set up block" function does not need to graduate into how to calculate type comments.

Also, a "smell" is that this function does nothing if you say `append_type_comment ` and you don't pass a `Node`, which is weird.

**Review by pablogsal — APPROVED** [review]

**Pull request merged**